src/packagers/JSConcatPackager.js

```diff
@@ -23,6 +24,31 @@ const helpers =
     .readFileSync(path.join(__dirname, '../builtins/helpers.js'), 'utf8')
     .trim() + '\n';
 
+async function hasSideEffects(asset, sideEffects) {
+  switch (typeof sideEffects) {
+    case 'undefined':
+      return true;
+    case 'boolean':
+      return sideEffects;
+    case 'string':
+      if (process.platform === 'win32') {
+        sideEffects = sideEffects.replace(/\\/g, '/');
+      }
+      return mm.isMatch(
+        path.relative((await asset.getPackage()).pkgdir, asset.name),
```

Comment:
You can save some I/O and `async` overhead by setting `pkgdir` in `cacheData` preemptively in `hoist.js` like we do for `sideEffects` :
https://github.com/parcel-bundler/parcel/blob/7423541be5bb5974e13862190ba82899ebaa0367/src/scope-hoisting/hoist.js#L34-L38